edx_event_bus_kafka/internal/consumer.py

```diff
@@ -392,6 +402,32 @@ def _check_receiver_results(self, send_results: list):
                 errors
             )
 
+    def _log_message_received(self, msg):
+        """
+        Log that a message was received, for audit log purposes.
+
+        See ADR: docs/decisions/0010_audit_logging.rst
+
+        This will not be sufficient to reconstruct the message; it will just
+        be enough to establish a timeline during debugging and to find a message
+        by offset.
+        """
+        try:
+            (ts_type, timestamp_ms) = msg.timestamp()
+            if ts_type == TIMESTAMP_NOT_AVAILABLE:
+                timestamp_info = "none"
+            else:
+                # Could be produce time or broker receive time; not going to bother to specify here.
+                timestamp_info = str(timestamp_ms)
+
+            logger.info(
```

Comment:
Hmm... do you think it's worth it? It would mean more header-reading code on the producer and consumer sides—not a _lot_ but enough that I would feel the need to add a new try-block around it on the producing side. Something like this:

```
            message_id = next(
                (h[1].decode() for h in msg.headers() if h[0] == HEADER_ID.message_header_key),
                None
            )
```

I guess I'm not actually sure what the role of that field is, since topic+partition+offset is already a unique key we can use to match up and identify messages. Maybe it matters more if we use the outbox pattern, or have DLQs?

Reply:
Related to our retro conversation, how about one of you write up a ticket for this instead (if you wish), and drop it on to the future milestone? Or, we can add it when and if we need it. It seems we have a start at audit logs either way.

Reply:
@timmc-edx @rgraber: Thoughts on my suggestion? Wondering how we can close this out. Thanks.